### PR TITLE
SON related operations, cli_wallet commands and RPC

### PR DESCRIPTION
**Pull request by obucinac**
- create_son, update_son, delete_son, list_sons
- get_sons, get_son_by_account, lookup_son_accounts, get_son_count

### CODE REVIEW

**Comment by obucinac** [comment]
Double check needed from here:
https://github.com/peerplays-network/peerplays/pull/160/commits/457c8b9d9f8ded0f06d7c84fd11e33369622dd2c

File: libraries/chain/db_maint.cpp [hunk 1]
```diff
@@ -326,6 +326,96 @@ void database::update_active_committee_members()
...
+      gp.active_sons.clear();
+      gp.active_sons.reserve(sons.size());
+      std::transform(sons.begin(), sons.end(),
+                     std::inserter(gp.active_sons, gp.active_sons.end()),
+                     [](const son_object& s) {
+         return s.id;
+      });
+   });
+
+   //const witness_schedule_object& wso = witness_schedule_id_type()(*this);
```
Comment (obucinac) [inline libraries/chain/db_maint.cpp]: Check this

File: libraries/chain/include/graphene/chain/config.hpp [hunk 2]
```diff
@@ -173,6 +175,8 @@
 #define GRAPHENE_PROXY_TO_SELF_ACCOUNT (graphene::chain::account_id_type(5))
 ///
 #define GRAPHENE_RAKE_FEE_ACCOUNT_ID (graphene::chain::account_id_type(6))
+///
+#define GRAPHENE_SON_ACCOUNT_ID (graphene::chain::account_id_type(7))
```
Comment (obucinac) [inline libraries/chain/include/graphene/chain/config.hpp]: Check what should be propper parameter. 7 is just selected

**Comment by oxarbitrage** [comment]
Just for the future, if you can add some meaningful names to commits instead of "WIP, SON operations, cli_wallet commands and RPC" to all of them will be much more clearer. Appreciated.

**Comment by obucinac** [comment]
We'll do squash merge, so commit names are not important, just details I will write in the end.

File: libraries/chain/db_maint.cpp [hunk 3]
```diff
@@ -326,6 +326,96 @@ void database::update_active_committee_members()
    });
 } FC_CAPTURE_AND_RETHROW() }
 
+void database::update_active_sons()
+{ try {
+   assert( _son_count_histogram_buffer.size() > 0 );
+   share_type stake_target = (_total_voting_stake-_witness_count_histogram_buffer[0]) / 2;
```
Comment (satyakoneru) [inline libraries/chain/db_maint.cpp]: it should be **_son_count_histogram_buffer**

[the remark this answers is not in the document]
Comment (obucinac) [reply libraries/chain/db_maint.cpp]: The logic behind this is that both update_active_comitee_members also contains _witness_count_histogram_buffer. It was weird, but I left it anyway.
Ill change it, and we'll see if some functinoality is different.

[the remark this answers is not in the document]
Comment (satyakoneru) [reply libraries/chain/db_maint.cpp]: update_active_comitee_members should also be changed I guess to _committee_count_histogram_buffer. But we can check that later.

[the remark this answers is not in the document]
Comment (obucinac) [reply libraries/chain/db_maint.cpp]: done

**Pull request merged**